src/main/kotlin/app/revanced/meta/readme/Semver.kt

```diff
@@ -0,0 +1,24 @@
+data class Semver(val major: Int, val minor: Int, val patch: Int) {
+    companion object {
+        fun fromString(version: String): Semver {
+            var parts = version.split(".")
+
+            if (parts.count() != 3) throw IllegalArgumentException("semver must have 3 parts")
+
+            val intParts = parts.map { it.toInt() }
+            return Semver(intParts[0], intParts[1], intParts[2])
+        }
+    }
+
+    override fun toString(): String = "$major.$minor.$patch"
+}
+
+class SemverComparator {
+    companion object : Comparator<Semver> {
```

Comment:
Instead of using a companion object here, simply convert the class to an `object`. It will be compiled into a singleton, where a companion object is a singleton inside the class.